src/java/org/apache/cassandra/db/marshal/PartitionerDefinedOrder.java

```diff
@@ -88,7 +93,8 @@ public Term fromJSONObject(Object parsed)
     @Override
     public String toJSONString(ByteBuffer buffer, ProtocolVersion protocolVersion)
     {
-        throw new UnsupportedOperationException();
+        assert baseType != null && !baseType.equals(this) : "PartitionerDefinedOrder's toJSONString method need a baseType but now is null or with a not euqal type.";
```

Comment:
Nit: The second part of this assertion does not make sense.

There is no way to construct a type for which `baseType == this`, because `baseType` is passed before `this` is constructed.

Reply:
So I just should remove this second assertion